### PR TITLE
Fixes for glGenerateMipmap

### PR DESCRIPTION
**Pull request by ben-clayton**
Fixes: #1312

### CODE REVIEW

**Comment by ben-clayton** [comment]
@dsrbecky - PTAL. I'd like to get this into V0.9 so I can start chasing zooshi folks for documentation.

File: gapis/api/gles/dependency_graph_behaviour_provider.go [hunk 1]
```diff
@@ -215,6 +215,22 @@ func (*GlesDependencyGraphBehaviourProvider) GetBehaviourForAtom(
...
+			case *GlGenerateMipmap:
+				tex, err := subGetBoundTextureOrErrorInvalidEnum(ctx, cmd, id, nil, s, GetState(s), cmd.Thread(), nil, cmd.Target)
+				if err != nil {
+					log.E(ctx, "Can not find bound texture %v", cmd.Target)
+				}
+				for levelIndex, level := range tex.Levels.Range() {
+					for layerIndex := range level.Layers.Range() {
+						data, size := tex.dataAndSize(levelIndex, layerIndex)
+						if levelIndex == 0 {
+							b.Read(g, data)
```
Comment (dsrbecky) [inline gapis/api/gles/dependency_graph_behaviour_provider.go]: Read both data and size

File: gapis/api/gles/dependency_graph_behaviour_provider.go [hunk 2]
```diff
@@ -215,6 +215,22 @@ func (*GlesDependencyGraphBehaviourProvider) GetBehaviourForAtom(
...
+						} else {
+							b.Read(g, size)
+							b.Write(g, data)
```
Comment (dsrbecky) [inline gapis/api/gles/dependency_graph_behaviour_provider.go]: Write both data and size

File: gapis/api/gles/dependency_graph_behaviour_provider.go (hunk 1 above)
Comment (ben-clayton) [reply gapis/api/gles/dependency_graph_behaviour_provider.go]: Okay

File: gapis/api/gles/dependency_graph_behaviour_provider.go (hunk 2 above)
Comment (ben-clayton) [reply gapis/api/gles/dependency_graph_behaviour_provider.go]: Does it write to size?

File: gapis/api/gles/dependency_graph_behaviour_provider.go (hunk 1 above)
Comment (ben-clayton) [reply gapis/api/gles/dependency_graph_behaviour_provider.go]: Done

File: gapis/api/gles/dependency_graph_behaviour_provider.go (hunk 2 above)
Comment (ben-clayton) [reply gapis/api/gles/dependency_graph_behaviour_provider.go]: Writing to size breaks tests and actual texture reads.

Comment (dsrbecky) [reply gapis/api/gles/dependency_graph_behaviour_provider.go]: I would say it writes to size, yes.  The spec is not 100% clear though.

"Mipmap generation replaces texture image levels levelbase + 1 through q with
images derived from the levelbase image, regardless of their previous contents."

My interpretation is that the previous non-zero levels are discarded (not-read), and completely replace (defined data+size).  In particular, my reading is that it is valid for the previous definitions of the images to be of wrong size, or absent at all.  Consequently, DCE should be killing any definitions of non-zero levels.

**Pull request merged**